---
title: "GEN349: Introduction to scRNA-seq"
author: "Meeta Mistry, Noor Sohail"
date: "January 16th, 2025"
---

# Schedule

## Pre-reading

* [Introduction to scRNA-seq](../lessons/01_intro_to_scRNA-seq.md)
* [Raw data to count matrix](../lessons/02_SC_generation_of_count_matrix.md)
* [Download this project](https://www.dropbox.com/s/vop78wq76h02a2f/single_cell_rnaseq.zip?dl=1)

## Day 1

| Time |  Topic  | Instructor |
|:-----------:|:----------:|:--------:|
| 13:00 - 13:15 | [Workshop introduction](../slides/workshop_intro_slides.pdf) | Meeta |
| 13:15 - 13:45 | scRNA-seq pre-reading discussion | All |
| 13:45 - 13:50 | Break |  |
| 13:50 - 14:20 | [Quality control of Cellranger counts]((../lessons/04_cellranger_QC.md)) | Noor |
| 14:20 - 14:55 | [Quality control setup in R]((../lessons/03_SC_quality_control-setup.md)) | Meeta |
| 14:55 - 15:00 | Overview of self-learning materials and homework submission | Meeta |


### Before the next class:

I. Please **study the contents** and **work through all the code** within the following lessons:
 
 1.  [Quality control with additional metrics](../lessons/04_SC_quality_control.md)
      <details>
       <summary><i>Click here for a preview of this lesson</i></summary>
         <br>In addition to the QC generated by cellranger, we can also compute some of our own metrics based on the raw data we have loaded into our Seurat object. <br><br>In this lesson you will:<br>
             - Compute essential QC metrics for each sample<br>
             - Create plots to visualize metrics across cells per sample<br>
             - Critically evaluate each plot and learn what each QC metric means<br><br>
        </details>
     
   2. [Theory of PCA](../lessons/05_theory_of_PCA.md)
      <details>
       <summary><i>Click here for a preview of this lesson</i></summary>
         <br>Before we can begin the next steps of the workflow, we need to make sure you have a good understanding of Principal Components Analysis (PCA). This method will be utilized in the scRNA-seq analysis workflow, and this foundation will help you better navigate those steps and interpretation of results.<br><br>
        </details>
         

II. **Submit your work**:
   * Each lesson above contains exercises; please go through each of them.
   * **Submit your answers** to the exercises using [this Google form](https://docs.google.com/forms/d/e/1FAIpQLSf4zw37eUcuNCmHxCLikg3JENoxGWIuv5c_5pqHdYNId81Arg/viewform) on **the day *before* the next class**.
   


### Questions?
* ***If you get stuck due to an error*** while runnning code in the lesson, [email us](mailto:hbctraining@hsph.harvard.edu) 

***

## Day 2

| Time |  Topic  | Instructor |
|:-----------:|:----------:|:--------:|
| 13:00 - 13:30 | Self-learning lessons discussion | All |
| 13:30 - 14:30|  [Normalization and regressing out unwanted variation](../lessons/06_SC_SCT_normalization.md) | Noor |
| 14:30 - 14:35 | Break |
| 14:35 - 15:00| [A brief introduction to Integration](../lessons/06a_integration_cca_theory.md) | Meeta |

### Before the next class:
I. Please **study the contents** and **work through all the code** within the following lessons:

1. [Running CCA integration and complex integration tasks](../lessons/06b_integration_code_harmony.md)
      <details>
       <summary><i>Click here for a preview of this lesson</i></summary>
        <br>In class, we described the theory of integration and in what situations we would implement it. <br><br>In this lesson you will:<br>
             - Run the code to implement CCA integration <br>
             - Evaluate the effect of integration on the UMAP <br>
             - Learn about methods for complex integration tasks (Harmonizing samples) <br>
        </details>
   
2. [Clustering](../lessons/07_SC_clustering_cells_SCT.md)
      <details>
       <summary><i>Click here for a preview of this lesson</i></summary>
         <br>From the UMAP visualization of our data  we can see that the cells are positioned into groups. Our next task is to isolate clusters of cells that are most similar to one another based on gene expression. <br><br>In this lesson you will:<br>
             - Learn the theory behind clustering and how it is performed in Seurat<br>
             - Cluster cells and visualize them on the UMAP<br>
        </details>

3. [Clustering quality control](../lessons/08_SC_clustering_quality_control.md)
      <details>
       <summary><i>Click here for a preview of this lesson</i></summary>
         <br>After separating cells into clusters, it is crtical to evaluate whether they are biologically meaningful or not. At this point we can also decide if we need to re-cluster and/or potentialy go back to a previous QC step.
         <br><br>In this lesson you will:<br>
           - Check to see that clusters are not influenced by uninteresting sources of variation<br>
           - Check to see whether the major principal components are driving the different clusters<br>
           - Explore the cell type identities by looking at the expression for known markers across the clusters.<br>
        </details>


II. **Submit your work**:
   * Each lesson above contains exercises; please go through each of them.
   * **Submit your answers** to the exercises using [this Google form](https://docs.google.com/forms/d/e/1FAIpQLSd8-k5YwfQekQQdFuDVPhGCy-eK70CqPiUkDNYDeWSEQYBAZg/viewform?usp=sf_link) on **the day *before* the next class**.

### Questions?
* ***If you get stuck due to an error*** while runnning code in the lesson, [email us](mailto:hbctraining@hsph.harvard.edu) 


***


## Day 3

| Time |  Topic  | Instructor |
|:-----------:|:----------:|:--------:|
| 13:00 - 13:45 | Self-learning lessons discussion | All |
| 13:45 - 14:30 |[Marker identification](../lessons/09_merged_SC_marker_identification.md) | Noor |
| 14:30 - 14:35 | Break |
| 14:35 - 14:50 | [Workflow summary and Q&A](../lessons/scRNAseq_workflow.md) | Meeta |
| 14:50- 15:00 | [Wrap up](../slides/Workshop_wrapup.pdf) | Meeta |

***

## Answer Keys

* **[Answer key - assignment #1](../homework/Day1_exercise_answer_key.md)**
* **[Answer key - assignment #2](../homework/Day2_exercise_answer_key.R)**

## Helpful HBC resources

* [Seurat Cheatsheet](../lessons/seurat_cheatsheet.md) 
* [Approaches for Differential expression analysis of scRNA-seq](https://hbctraining.github.io/DGE_analysis_scRNAseq/)

***

## Resources
We have covered the analysis steps in quite a bit of detail for scRNA-seq exploration of cellular heterogeneity using the Seurat package. For more information on topics covered, we encourage you to take a look at the following resources:

### Seurat-focused
* [Seurat vignettes](https://satijalab.org/seurat/vignettes.html)
* [Seurat cheatsheet](https://satijalab.org/seurat/essential_commands.html)
* [Satija Lab: Single Cell Genomics Day](https://satijalab.org/scgd21/)
* [Additional information about cell cycle scoring](../lessons/cell_cycle_scoring.md)

### Scaling up: scRNA-seq analysis on HPC  
* Using RStudio on O2
    * [HMSRC wiki page](https://harvardmed.atlassian.net/wiki/spaces/O2/pages/1623425967/RStudio+on+O2)
    * [HBC RStudio on O2 tutorial](https://hbctraining.github.io/Intro-to-Unix-QMB/lessons/R_studio_on_02.html)

### Cell type annotation
- Databases with markers for manual annotation
  - [CellMarker 2.0](http://bio-bigdata.hrbmu.edu.cn/CellMarker/)
  - Cell type signature gene sets from [MSigDb](https://www.gsea-msigdb.org/gsea/msigdb/human/genesets.jsp?collection=C8)
  - [CELL x GENE from CZI](https://cellxgene.cziscience.com/gene-expression)
- Reference-based automated celltype annotation
  - [Azimuth](https://azimuth.hubmapconsortium.org/)
  - [Celltypist](https://www.celltypist.org/)
 
   
### Highlighted papers

- ["Sampling time-dependent artifacts in single-cell genomics studies."](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-02032-0) *Massoni-Badosa et al.* 2019
- ["Dissociation of solid tumor tissues with cold active protease for single-cell RNA-seq minimizes conserved collagenase-associated stress responses."](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1830-0) *O'Flanagan et al.* 2020
- ["Systematic assessment of tissue dissociation and storage biases in single-cell and single-nucleus RNA-seq workflows."](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-02048-6) *Denisenko et al.* 2020
- ["Confronting false discoveries in single-cell differential expression", _Nature Communications_ 2021](https://www.nature.com/articles/s41467-021-25960-2)
- [Single-nucleus and single-cell transcriptomes compared in matched cortical cell types](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6306246/)
- [A single-cell and single-nucleus RNA-Seq toolbox for fresh and frozen human tumors](https://www.nature.com/articles/s41591-020-0844-1)
- [Ligand-receptor analysis with CellphoneDB](https://www.nature.com/articles/s41576-020-00292-x)
- [Best practices for single-cell analysis across modalities](https://www.nature.com/articles/s41576-023-00586-w)



### Other online scRNA-seq courses:
  - [OSCA with Bioconductor](http://bioconductor.org/books/release/OSCA/)
  - [DFCI/Shirley Liu](https://liulab-dfci.github.io/bioinfo-combio/)
  - [Wellcome Sanger Institute/Hemmberg Lab](https://www.singlecellcourse.org/)
  - [ISCB Workshop](https://github.com/SingleCellTranscriptomics)
  - [Broad workshop](https://broadinstitute.github.io/2020_scWorkshop/)
  - [SciLifeLab workshop](https://nbisweden.github.io/workshop-scRNAseq/)



